code/uilib/uipulldownmenucontainer.cpp

```diff
@@ -22,14 +22,138 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+);
+
+Event EV_UIPulldownMenuContainer_PopupHighlightFGColor
+(
+	"popup_highlight_fgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the highlight background color of the popup menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
+Event EV_UIPulldownMenuContainer_HighlightBGColor
+(
+	"highlight_bgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the highlight foreground color of the pulldown menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
+Event EV_UIPulldownMenuContainer_HighlightFGColor
+(
+	"highlight_fgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the highlight background color of the pulldown menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
+Event EV_UIPulldownMenuContainer_PopupHighlightBGColor
+(
+	"popup_highlight_bgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the highlight foreground color of the popup menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
+Event EV_UIPulldownMenuContainer_PopupFGColor
+(
+	"popup_fgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the background color of the popup menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
+Event EV_UIPulldownMenuContainer_PopupBGColor
+(
+	"popup_bgcolor",
+	EV_DEFAULT,
+	"sffff",
+	"menu r g b a",
+	"Set the foreground color of the popup menu\n"
+	"menu - the name of the popup menu to color\n"
+);
+
 CLASS_DECLARATION( UIWidget, UIPulldownMenuContainer, NULL )
 {
+	{ &EV_UIPulldownMenuContainer_AddPopup,					&UIPulldownMenuContainer::AddPopup					},
+	{ &EV_UIPulldownMenuContainer_MenuShader,				&UIPulldownMenuContainer::SetMenuShader				},
+	{ &EV_UIPulldownMenuContainer_SelectedMenuShader,		&UIPulldownMenuContainer::SetSelectedMenuShader		},
+	{ &EV_UIPulldownMenuContainer_PopupHighlightFGColor,	&UIPulldownMenuContainer::SetPopupHighlightFGColor	},
+	{ &EV_UIPulldownMenuContainer_PopupHighlightBGColor,	&UIPulldownMenuContainer::SetPopupHighlightBGColor	},
+	{ &EV_UIPulldownMenuContainer_PopupFGColor,				&UIPulldownMenuContainer::SetPopupFGColor			},
+	{ &EV_UIPulldownMenuContainer_PopupBGColor,				&UIPulldownMenuContainer::SetPopupBGColor			},
+	{ &EV_UIPulldownMenuContainer_HighlightFGColor,			&UIPulldownMenuContainer::SetHighlightFGColor		},
+	{ &EV_UIPulldownMenuContainer_HighlightBGColor,			&UIPulldownMenuContainer::SetHighlightBGColor		},
+	{ &EV_Layout_VirtualRes,								&UIPulldownMenuContainer::PulldownVirtualRes		},
 	{ NULL, NULL }
 };
 
 UIPulldownMenuContainer::UIPulldownMenuContainer()
 {
-	// FIXME: stub
+	m_menu = new UIPulldownMenu();
+	m_menu->CreateAligned(this, this);
+}
+
+
+UIPulldownMenuContainer::~UIPulldownMenuContainer()
+{
+	for (int i = m_popups.NumObjects(); i > 0; i--)
+	{
+		uipopup_describe* uid = m_popups.ObjectAt(i);
+		m_popups.RemoveObjectAt(i);
+		delete uid;
+	}
+
+	for (int i = m_dataContainer.NumObjects(); i > 0; i--)
+	{
+		char* data = m_dataContainer.ObjectAt(i);
+		m_dataContainer.RemoveObjectAt(i);
+		delete data;
+	}
```

Comment:
`m_menu` doesn't ever seem to be freed, could be a memleak.

Reply:
Add a FIXME: comment to avoid forgetting about it